core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -802,25 +812,60 @@ private String buildConfPath(Map<String, String> properties, String provider) {
     return confPath;
   }
 
-  private String buildPkgPath(Map<String, String> conf, String provider) {
+  private String buildPkgPath(
+      Map<String, String> conf, String provider, IsolatedClassLoader.IsolatedType isolatedType) {
     String gravitinoHome = System.getenv("GRAVITINO_HOME");
     Preconditions.checkArgument(gravitinoHome != null, "GRAVITINO_HOME not set");
     boolean testEnv = System.getenv("GRAVITINO_TEST") != null;
 
     String pkg = conf.get(Catalog.PROPERTY_PACKAGE);
-    String pkgPath;
     if (pkg != null) {
-      pkgPath = String.join(File.separator, pkg, "libs");
-    } else if (testEnv) {
-      // In test, the catalog package is under the build directory.
-      pkgPath =
-          String.join(
-              File.separator, gravitinoHome, "catalogs", "catalog-" + provider, "build", "libs");
-    } else {
-      // In real environment, the catalog package is under the catalog directory.
-      pkgPath = String.join(File.separator, gravitinoHome, "catalogs", provider, "libs");
+      return String.join(File.separator, pkg, "libs");
```

Comment:
How do we handle this case if `Catalog.PROPERTY_PACKAGE` is set? From the current code, if `Catalog.PROPERTY_PACKAGE` is set, then we'll directly return without handling authorization related packages.

Reply:
I refactor these codes.